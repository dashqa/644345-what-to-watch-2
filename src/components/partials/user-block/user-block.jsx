import React from "react";
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import PropTypes from "prop-types";

import {BASE_URL} from "@api/constants";
import {getUser} from "@store/user-data/selectors";

const UserBlock = ({user}) => {
  const {name, avatarImg} = user;
  const imgPath = (avatarImg || ``).split(`/wtw`).join(``);
  return (
    <div className="user-block">
      {Object.keys(user).length ? (
        <div className="user-block__avatar">
          <Link to={`/my-list`}>
            <img
              src={`${BASE_URL}${imgPath}`}
              alt={name}
              width="63"
              height="63"
            />
          </Link>
        </div>
      ) : (
        <Link to={`/login`} className="user-block__link">Sign in</Link>
      )}
    </div>
  );
};

UserBlock.defaultProps = {
  user: null,
};

UserBlock.propTypes = {
  user: PropTypes.shape({
    name: PropTypes.string,
    avatarImg: PropTypes.string,
  })
};

const mapStateToProps = (state) => ({
  user: getUser(state),
});

export {UserBlock};

export default connect(mapStateToProps)(UserBlock);
