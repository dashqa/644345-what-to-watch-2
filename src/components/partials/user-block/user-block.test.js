import React from 'react';
import renderer from 'react-test-renderer';
import {UserBlock} from './user-block';
import {BrowserRouter} from "react-router-dom";

const props = {
  user: {
    name: `Dasha`,
    avatar: `https://image.com/image.png`
  }
};

it(`User block correctly renders after relaunch`, () => {
  const tree = renderer.create(<BrowserRouter><UserBlock {...props}/></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
