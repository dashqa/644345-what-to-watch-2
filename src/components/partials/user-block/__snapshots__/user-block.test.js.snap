// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User block correctly renders after relaunch 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <a
      href="/my-list"
      onClick={[Function]}
    >
      <img
        alt="Dasha"
        height="63"
        src="https://htmlacademy-react-2.appspot.com/wtw"
        width="63"
      />
    </a>
  </div>
</div>
`;
